### PR TITLE
Added queue stat counters map

### PR DESCRIPTION
**Pull request by andrii-savka**
SWSS part of SNMP queue stat counters feature.

### CODE REVIEW

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -49,6 +52,9 @@ PortsOrch::PortsOrch(DBConnector *db, vector<string> tableNames) :
     /* Initialize port table */
     m_portTable = unique_ptr<Table>(new Table(m_db, APP_PORT_TABLE_NAME));
 
+    /* Initialize queue table */
+    m_queueTable = unique_ptr<Table>(new Table(counter_db, COUNTERS_QUEUE_NAME_MAP));
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: try make_unique?
If it works, please help modify existing code above.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -1045,7 +1051,21 @@ void PortsOrch::initializeQueues(Port &port)
         SWSS_LOG_ERROR("Failed to get queue list for port %s rv:%d", port.m_alias.c_str(), status);
         throw runtime_error("PortsOrch initialization failure.");
     }
+
     SWSS_LOG_INFO("Get queues for port %s", port.m_alias.c_str());
+
+    /* Create the Queue map in the Counter DB */
+    vector<FieldValueTuple> vector;
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >vector [](start = 28, length = 6)

vector as variable name is really confusing.

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -1045,7 +1051,21 @@ void PortsOrch::initializeQueues(Port &port)
...
+
+    for (size_t queue_index = 0; queue_index < port.m_queue_ids.size(); ++queue_index)
+    {
+        std::stringstream name;
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: ostringstream

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -1045,7 +1051,21 @@ void PortsOrch::initializeQueues(Port &port)
...
+        name << port.m_alias << ":" << queue_index;
+        FieldValueTuple tuple(name.str(), sai_serialize_object_id(port.m_queue_ids[queue_index]));
+        vector.push_back(tuple);
+    }
+
+    m_queueTable->set("", vector);
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: design question: why serialize all pairs into 'vector'? why not store them as key-value in the table?

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (marian-pritsak) [reply orchagent/portsorch.cpp]: Why do we need to mix refactoring with new functionality? This line is written in the same style as above. If there's a need for this kind of change, it should be done separately.

Comment (qiluo-msft) [reply orchagent/portsorch.cpp]: I don't know @andriis@mellanox.com 's idea. If you think the code review comment is not acceptable, I am ok with new iteration only on the new code. I prefer better code quality than consistent code style.

Let me handle existing code refactoring if you don't want to.

---
In reply to: [141885713](https://github.com/Azure/sonic-swss/pull/323#discussion_r141885713) [](ancestors = 141885713)

File: orchagent/portsorch.cpp (hunk 4 above)
Comment (andrii-savka) [reply orchagent/portsorch.cpp]: It's written in the same style as port counters

**Comment by andrii-savka** [comment]
Fixed review remarks.

**Pull request merged**